packages/instant/src/redux/reducer.ts

```diff
@@ -121,3 +130,24 @@ export const reducer = (state: State = INITIAL_STATE, action: Action): State =>
             return state;
     }
 };
+
+const doesBuyQuoteMatchState = (buyQuote: BuyQuote, state: State): boolean => {
+    const selectedAssetIfExists = state.selectedAsset;
+    const selectedAssetAmountIfExists = state.selectedAssetAmount;
+    // if no selectedAsset or selectedAssetAmount exists on the current state, return false
+    if (_.isUndefined(selectedAssetIfExists) || _.isUndefined(selectedAssetAmountIfExists)) {
+        return false;
+    }
+    // if buyQuote's assetData does not match that of the current selected asset, return false
+    if (selectedAssetIfExists.assetData !== buyQuote.assetData) {
+        return false;
+    }
+    // if buyQuote's assetBuyAmount does not match selectedAssetAmount, return false
+    const selectedAssetMetaData = selectedAssetIfExists.metaData;
+    const selectedAssetAmountBaseUnits =
+        selectedAssetMetaData.assetProxyId === AssetProxyId.ERC20
+            ? Web3Wrapper.toBaseUnitAmount(selectedAssetAmountIfExists, selectedAssetMetaData.decimals)
+            : new BigNumber(1);
```

Comment:
Does ERC721 have the concept of base units? I guess the `BuyQuote` would just have the correct amount? This doesn't seem to cover the case where the user types "0" but we're not planning to show an input and automatically put selected amount to 1 I guess.

Reply:
I'm going to amend this to be more flexible, right now we're making too many assumptions about the nature of future non erc20 assets